### PR TITLE
Fetch api

### PR DESCRIPTION
**Pull request by sanika621879**

### CODE REVIEW

File: apidoc.yaml [hunk 1]
```diff
@@ -0,0 +1,567 @@
+openapi: 3.0.1
+info:
+  title: GenKwiz API
+  description: 'Interactive quizzing API with questions of multiple engrossing genres.  '
```
Comment (rdsarda) [inline apidoc.yaml:4]: Minor : no quotes required at the start and end, remove extra spaces at the end after full stop

File: apidoc.yaml [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+  contact:
+    name: Shriram Deshpande
+    url: 'http://genkwiz.com'
+    email: deshpandeshriram@gmail.com
+  license:
+    name: API License
+    url: 'http://genkwiz.com'
+  version: 0.0.2-oas3
```
Comment (rdsarda) [inline apidoc.yaml:12]: Should be `0.0.2`

File: apidoc.yaml [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+servers:openapi: 3.0.1
+info:
```
Comment (rdsarda) [inline apidoc.yaml:14]: The info element and its child elements appear twice, pls remove this instance.

File: api-spec.yaml [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+        '201':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  id:
+                    description: quiz id .
+                    type: integer
```
Comment (rdsarda) [inline api-spec.yaml:57]: Change quizId to be an UUID. If we keep it as an integer, it becomes easy for everyone to guess quizIds, and users can just make API calls to fetch quizzes that were created for other users.

Note: We'll need to make a corresponding change in DB script as well.

File: api-spec.yaml [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+        '404':
+          description: Not Found
+          content: {}
+      deprecated: false
+   
+  '/answers/{quizId}':
+    post:
+      tags:
+        - quiz-controller
+      description: submit and display correct answer .
```
Comment (vrushbru) [inline api-spec.yaml:121]: Description to should be more elaborative. Follow that throughout the doc

File: api-spec.yaml [hunk 6]
```diff
@@ -0,0 +1,476 @@
...
+          description: Forbidden
+          content: {}
+        '404':
+          description: Not Found
+          content: {}
+        '500':
+          description: Internal Server Error
+          content: {} 
+          
+  '/performance/{quizId}':
```
Comment (vrushbru) [inline api-spec.yaml:185]: no need of this endpoint ..task can be done at the backend side.

File: api-spec.yaml [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+            format: int32
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Quiz'
```
Comment (vrushbru) [inline api-spec.yaml:105]: response for quiz to be defined as array of individual required elements only.

File: api-spec.yaml [hunk 8]
```diff
@@ -0,0 +1,476 @@
...
+          required: true
+          schema:
+            type: string
+        - name: type
+          in: query
+          description: Type of the Question (MCQ/Subjective)
+          required: true
+          schema:
+            type: string    
+        - name: params
```
Comment (vrushbru) [inline api-spec.yaml:142]: this should be request body rather than query parameters

File: api-spec.yaml [hunk 9]
```diff
@@ -0,0 +1,476 @@
...
+        Ratings:
+          type: number
+          format: float
+        Type:
+          type: string
+    Answer:
+      title: Answer of the question.
+      type: object
+      properties:
+        questionId:
```
Comment (vrushbru) [inline api-spec.yaml:416]: the entire question object is not needed.keep the question id hint

File: api-spec.yaml [hunk 10]
```diff
@@ -0,0 +1,476 @@
...
+          description: Forbidden
+          content: {}
+        '404':
+          description: Not Found
+          content: {}
+        '500':
+          description: Internal Server Error
+          content: {}  
+          
+  '/score':
```
Comment (vrushbru) [inline api-spec.yaml:234]: rename this end point to quizzes/quizid/submit

File: api-spec.yaml (hunk 4 above)
Comment (vrushbru) [reply api-spec.yaml:57]: done

File: api-spec.yaml (hunk 5 above)
Comment (vrushbru) [reply api-spec.yaml]: done

File: api-spec.yaml (hunk 6 above)
Comment (sanika621879) [reply api-spec.yaml]: done

File: api-spec.yaml (hunk 7 above)
Comment (sanika621879) [reply api-spec.yaml:105]: Point to be discussed.

File: api-spec.yaml (hunk 8 above)
Comment (sanika621879) [reply api-spec.yaml]: done

File: api-spec.yaml (hunk 9 above)
Comment (sanika621879) [reply api-spec.yaml:326]: done

File: api-spec.yaml (hunk 10 above)
Comment (sanika621879) [reply api-spec.yaml]: done

File: api-spec.yaml (hunk 7 above)
Comment (vrushbru) [reply api-spec.yaml:105]: done

**Pull request closed without merging**